workflow/scripts/get-transcript-info.R

```diff
@@ -39,69 +40,88 @@ while ( class(mart)[[1]] != "Mart" ) {
+}else if (has_canonical) {
+     attributes <- c(attributes, "transcript_is_canonical")
 }
-
 t2g <- biomaRt::getBM(
-            attributes = attributes,
-            mart = mart,
-            useCache = FALSE
-            )
-if (!has_canonical) {
-  t2g <- t2g %>% add_column(transcript_is_canonical = NA)
+attributes = attributes,
+mart = mart,
+useCache = FALSE
+)
+# Set columns as NA if three_prime_activated is set to false or if the transcipts are not canonical
+if (!has_canonical || !three_prime_activated) {
+  t2g <- t2g %>% add_column(chromosome_name = NA, transcript_mane_select = NA,
+      ensembl_transcript_id_version = NA)
+}else if (!has_canonical) {
+   t2g <- t2g %>% add_column(transcript_is_canonical = NA)
 }
-
 t2g <- t2g %>%
-        rename( target_id = ensembl_transcript_id,
-                ens_gene = ensembl_gene_id,
-                ext_gene = external_gene_name,
-                gene_desc = description,
-                canonical = transcript_is_canonical
-                ) %>%
-        mutate_at(
-          vars(gene_desc),
-          function(values) { str_trim(map(values, function (v) { str_split(v, r"{\[}")[[1]][1]})) } # remove trailing source annotation (e.g. [Source:HGNC Symbol;Acc:HGNC:5])
-        ) %>%
-        mutate_at(
-          vars(canonical),
-          function(values) {
-            as_vector(
-              map(
-                str_trim(values), 
-                function(v) {
-                  if (is.na(v)) {
-                    NA
-                  } else if (v == "1") { 
-                    TRUE 
-                  } else if (v == "0") {
-                    FALSE
-                  }
-                }
-              )
-            )
+  rename(
+    target_id = ensembl_transcript_id,
+    ens_gene = ensembl_gene_id,
+    ext_gene = external_gene_name,
+    gene_desc = description,
+    canonical = transcript_is_canonical,
+    chromosome_name = chromosome_name,
+    transcript_mane_select = transcript_mane_select,
+    ensembl_transcript_id_version = ensembl_transcript_id_version,
+  ) %>%
+  mutate_at(
+    vars(gene_desc),
+    function(values) {
+      str_trim(map(values, function(v) {
+        str_split(v, r"{\[}")[[1]][1]
+      }))
+    } # remove trailing source annotation (e.g. [Source:HGNC Symbol;Acc:HGNC:5])
+  ) %>%
+  mutate_at(
+    vars(canonical,),
+    function(values) {
+      as_vector(
+        map(
+          str_trim(values),
+          function(v) {
+            if (is.na(v)) {
+              NA
+            } else if (v == "1") {
+              TRUE
+            } else if (v == "0") {
+              FALSE
+            }
           }
         )
-
+      )
+    }
+  )
+# Filter transcipts that are canonical, mane selected and filter chromosomes that are defined as "patch" 
+if (three_prime_activated && has_canonical) {
+  t2g <- t2g %>%
+    filter(!str_detect(chromosome_name, "patch|PATCH")) %>%
+    filter(str_detect(transcript_mane_select, ""))
```

Comment:
So, if we only use mane select here, why do we have the has_canonical stuff above? Shouldn't we rather check for mane_select? Also, doesn't three_prime_activated already imply that canonical/mane_select attributes are present? And if not, I think there should rather be an error saying that we need canonical/mane_select if three prime mode is activated.

Reply:
Noted. Yes and I have added and else if case if three_prime is activated and it mane_select is not present.